notebook/converted/benchmark_results.py

```diff
@@ -1313,3 +1315,143 @@ def return_unlabeled():
+
+
+in_files = glob.glob('../../results/sample-split.*.tsv')
+print(in_files[:5])
+
+
+# In[15]:
+
+
+tissue_metrics = pd.DataFrame()
+
+for path in in_files:
+    new_df = pd.read_csv(path, sep='\t')
+    model_info = path.strip('.tsv').split('biobert.')[-1]
+    model_info = model_info.split('.')
+    model_info = model_info[-1]
+    model_info = model_info.split('_')
+        
+    supervised_model = '_'.join(model_info[:2])
+             
+    new_df['supervised'] = supervised_model
+    
+    new_df['seed'] = model_info[-1]
+            
+    tissue_metrics = pd.concat([tissue_metrics, new_df])
+    
+tissue_metrics['train_count'] = tissue_metrics['train sample count']
+tissue_metrics['supervised'] = tissue_metrics['supervised'].str.replace('pytorch_supervised', 'three_layer_net')
+
+tissue_metrics
+
+
+# In[16]:
+
+
+plot = ggplot(tissue_metrics, aes(x='train_count', y='balanced_accuracy', color='is_pretrained')) 
+plot += geom_smooth()
+plot += geom_point(alpha=.2)
+plot += ggtitle('Transfer Learning Sample Positive Control')
+plot += facet_grid('supervised ~ .')
+plot
+
+
+# In[17]:
+
+
+single_run_df = tissue_metrics[tissue_metrics['seed'] == '1']
+single_run_df.head()
+
+
+# In[18]:
+
+
+plot = ggplot(single_run_df, aes(x='train_count', y='balanced_accuracy', color='is_pretrained')) 
+#plot += geom_smooth()
+plot += geom_point(alpha=.2)
+plot += ggtitle('Sample Control Single Run Points')
+plot += facet_grid('supervised ~ .')
+plot
+
+
+# In[ ]:
+
+
+
+
+
+# ## Study Split Positive Control
+
+# In[19]:
+
+
+in_files = glob.glob('../../results/study-split.*.tsv')
+print(in_files[:5])
+
+
+# In[20]:
+
+
+tissue_metrics = pd.DataFrame()
+
+for path in in_files:
+    new_df = pd.read_csv(path, sep='\t')
+    model_info = path.strip('.tsv').split('biobert.')[-1]
+    model_info = model_info.split('.')
+    model_info = model_info[-1]
+    model_info = model_info.split('_')
+        
+    supervised_model = '_'.join(model_info[:2])
+             
+    new_df['supervised'] = supervised_model
+    
+    new_df['seed'] = model_info[-1]
+            
+    tissue_metrics = pd.concat([tissue_metrics, new_df])
+    
+tissue_metrics['train_count'] = tissue_metrics['train sample count']
+tissue_metrics['supervised'] = tissue_metrics['supervised'].str.replace('pytorch_supervised', 'three_layer_net')
+
+tissue_metrics
+
+
+# In[21]:
+
+
+plot = ggplot(tissue_metrics, aes(x='train_count', y='balanced_accuracy', color='is_pretrained')) 
+plot += geom_smooth()
+plot += geom_point(alpha=.2)
+plot += ggtitle('Transfer Learning Study Positive Control')
+plot += facet_grid('supervised ~ .')
+plot
+
+
+# In[22]:
+
+
+single_run_df = tissue_metrics[tissue_metrics['seed'] == '1']
+single_run_df.head()
+
+
+# In[23]:
+
+
+plot = ggplot(single_run_df, aes(x='train_count', y='balanced_accuracy', color='is_pretrained')) 
+plot += geom_smooth()
+plot += geom_point(alpha=.2)
+plot += ggtitle('Study Level Single Run Points')
+plot += facet_grid('supervised ~ .')
+plot
```

Comment:
Could breaking the results down by tissue type help you understand the balanced accuracy 55% maximum? Maybe there are certain tissue types where performance is always poor, and these are effectively putting a cap on how high your accuracy can be overall - just an idea.

Reply:
That's a good question. I would expect logistic regression to have the same performance in both sets because you're optimizing a concave function with a global optimum. In that sense if training works correctly, the only thing the starting weight change is how long it takes to get there.

On the other hand, my pytorch implementation of logistic regression is helped by pretraining. Maybe the pytorch version isn't given enough epochs to converge, or (my working theory) that the sklearn optimizer is less noisy than batch SGD